api/qanpb/profile.proto

```diff
@@ -39,32 +40,37 @@ message ReportMapFieldEntry {
 
 // ReportReply is list of reports per quieryids, hosts etc.
 message ReportReply {
-  repeated ProfileRow rows = 1;
+  uint32 total_rows = 1;
+  uint32 offset = 2;
+  uint32 limit = 3;
+  repeated Row rows = 4;
 }
 
-// ProfileRow define metrics for selected dimention.
-message ProfileRow {
+// Row define metrics for selected dimention.
+message Row {
   uint32 rank = 1;
-  float percentage = 2;
-  string dimension = 3;
-  float row_number = 4;
-  string d_servers = 5;
-  string d_databases = 6;
-  string d_schemas = 7;
-  string d_usernames = 8;
-  string d_client_hosts = 9;
-  string first_seen = 10;
-  float qps = 11;
-  float load = 12;
-  string fingerprint = 13;
-  Stats stats = 14;
+  string dimension = 2;
+  map<string, Metric> metrics = 3;
+  repeated Point sparkline = 4;
 }
 
-// Stats metrics.
-message Stats {
-  float num_queries = 1;
-  float m_query_time_sum = 2;
-  float m_query_time_min = 3;
-  float m_query_time_max = 4;
-  float m_query_time_p99 = 5;
+// Metric cell.
+message Metric {
+  Stat stats = 1;
+}
```

Comment:
Do we really need that type with a single field? Will it grow?

Reply:
Yes, it will grow up with generic metrics like QPC, percent of total, etc.